tests/test_cli.py

```
@@ -453,6 +453,102 @@ def test_file_upload_closes_vm_on_failure(
+        self,
+        mock_vm_cls: MagicMock,
+        tmp_path: Path,
+        capsys: pytest.CaptureFixture,
+    ) -> None:
+        """`smolvm file download` should copy a guest file to the host."""
+        destination = tmp_path / "note.txt"
+        vm = MagicMock()
+        vm.download_file.return_value = str(destination)
+        mock_vm_cls.from_id.return_value = vm
+
+        ret = main(["file", "download", "vm001", "/tmp/note.txt", str(destination)])
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/note.txt"

(S108)